actions/class.ItemContent.php

```diff
@@ -36,29 +36,33 @@ class taoItems_actions_ItemContent extends tao_actions_CommonModule
 {
     use OntologyAwareTrait;
 
+    private const DEFAULT_PAGINATION_LIMIT = 10;
+    private const DEFAULT_PAGINATION_OFFSET = 0;
+
     /**
-     * Returns a json encoded array describign a directory
+     * Returns a json encoded array describing a directory
      *
      * @throws common_exception_MissingParameter
-     * @return string
      */
-    public function files()
+    public function files(): void
     {
-        if (!$this->hasRequestParameter('uri')) {
+        $params = $this->getPsrRequest()->getQueryParams();
+
+        if (!isset($params['uri'])) {
             throw new common_exception_MissingParameter('uri', __METHOD__);
         }
-        $itemUri = $this->getRequestParameter('uri');
-        $item = $this->getResource($itemUri);
+        $itemUri = $params['uri'];
+        $item = $this->getResource($params['uri']);
 
-        if (!$this->hasRequestParameter('lang')) {
+        if (!isset($params['lang'])) {
             throw new common_exception_MissingParameter('lang', __METHOD__);
         }
-        $itemLang = $this->getRequestParameter('lang');
+        $itemLang = $params['lang'];
```

Comment:
In your case you will call `getResource` but then if `lang` param is not exist - you will throw an error.
To avoid this we can move second check to the top and after both checks we will call `getResource` method.
Wdyt?
```
        if (!isset($params['uri'])) {
            throw new common_exception_MissingParameter('uri', __METHOD__);
        }

        if (!isset($params['lang'])) {
            throw new common_exception_MissingParameter('lang', __METHOD__);
        }

        $itemUri = $params['uri'];
        $item = $this->getResource($params['uri']);
        $itemLan = $params['lang'];
```